### ISSUE TITLE
Nodegroup role name exceeds maximum length

### ISSUE THREAD

**Issue by richardcase** [kubernetes-sigs/cluster-api-provider-aws#2704]
/kind bug
/area provider/eks
/priority important-soon
/milestone v0.7.x

**What steps did you take and what happened:**

Created stack to allow iam roles per cluster:

```yaml
apiVersion: bootstrap.aws.infrastructure.cluster.x-k8s.io/v1alpha1
kind: AWSIAMConfiguration
spec:
  bootstrapUser:
    enable: true
  eks:
    enable: true
    iamRoleCreation: true # Set to true if you plan to use the EKSEnableIAM feature flag to enable automatic creation of IAM roles
```

Enabled IAM roles per cluster using the following before `clusterctl init`:

```bash
export EXP_EKS_IAM=true"
```

Created a machine pool with the following specs:

```yaml
apiVersion: exp.cluster.x-k8s.io/v1alpha3
kind: MachinePool
metadata:
  name: "capi-managed-test-pool-0"
spec:
  clusterName: "capi-managed-test"
  template:
    spec:
      clusterName: "capi-managed-test"
      bootstrap:
        dataSecretName: ""
      infrastructureRef:
        name: "capi-managed-test-pool-0"
        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
        kind: AWSManagedMachinePool
---
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
kind: AWSManagedMachinePool
metadata:
  name: "capi-managed-test-pool-0"
spec:
  instanceType: t3.large
  amiType: AL2_x86_64
```

And then we get the following error when reconciliation occurs:

```bash
[manager] E0824 12:46:07.186195       8 controller.go:257] controller-runtime/controller "msg"="Reconciler error" "error"="failed to reconcile machine pool for AWSManagedMachinePool default/capi-managed-test-pool-0: ValidationError: 1 validation error detected: Value 'default_capi-managed-test-control-plane-default_capi-managed-test-pool-0-nodegroup-iam-service-role' at 'roleName' failed to satisfy constraint: Member must have length less than or equal to 64\n\tstatus code: 400, request id: d1f43a56-6b8f-4ebe-984c-6c24647e43e9" "controller"="awsmanagedmachinepool" "name"="capi-managed-test-pool-0" "namespace"="default"
```

**What did you expect to happen:**
I would expect the controller to detect the length of the auto-generated role name and then truncate/hash if its too long.

**Anything else you would like to add:**
This fix will need to be applied to 0.7x and then back ported to 0.6.x

**Environment:**

- Cluster-api-provider-aws version: 0.6.8

**Comment by Callisto13** [comment]
@richardcase can I take this one?

**Comment by Sajiyah-Salat** [comment]
Is this issue still needs to solve

**Comment by Callisto13** [comment]
@richardcase the description originally said that a fix should be backported to 0.6.0, but that is a very old version now. Do we want to bother doing that or close this as fixed from 0.7.0?

**Comment by richardcase** [comment]
As this is fixed in the currently supported versions: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/eks/roles.go#L165:L168 (thanks @Callisto13 for checking this) we can now close this issue as there is no need to backport to 0.6.0 as its unsupported now.
